drivers/flash/soc_flash_b91.c

```diff
@@ -104,6 +135,21 @@ static int flash_b91_write(const struct device *dev, off_t offset,
 		return -EINVAL;
 	}
 
+	/* need to store data in intermediate RAM buffer in case from flash to flash write */
+	if ((uint32_t)data > FLASH_ORIGIN) {
+
+		buf = k_malloc(len);
```

Comment:
I would add a limit for memory allocation size and do write of data bigger than data in a few chunks.
This will reduce possibility of encountering memory unavailable issues while writing for flash to flash.

Reply:
It makes sense.  However, I prefer to leave it as it is because even if check for CONFIG_HEAP_MEM_POOL_SIZE the k_malloc function still can return NULL, and users still need to do this chunking on the upper layer (or increase CONFIG_HEAP_MEM_POOL_SIZE).
Are you OK with it, or is it something that must be fixed?

Reply:
Yes, with the local variable approach the non-mem issue is resolved.
However, I still prefer to avoid chunking in the driver to avoid potential performance issues.

Reply:
Ok, you know the limitation.